OfflineUX.md

```diff
@@ -0,0 +1,129 @@
+
+We send a request to the API to keep track of that. After all, if the user signed in on a different device, we would want all of their pinned chats to show up (another way of saying this is we want their pinned chats to ‘persist’ across devices). 
+The chat should go to the top of the list with the other pinned chats, and the pin button should look darker than it did before. This is the user’s visual feedback that clicking the pin button worked.
+
+If the user is offline, do we need to wait for the API request to finish before doing all that visual stuff? No! We are optimistic that the API request will succeed once it’s sent. In the meantime, we let the user continue using the app. 
+</div>
+
+The example we just looked at is nice and simple, but what about something more complicated, like requesting money from another user? What about paying another user? Or even worse, removing a user from a workspace? For these types of actions, we can’t simply proceed as if the request already finished. That would have serious consequences for the user. This is why we have developed different ways of handling these cases. These are called the UX Patterns, which are explained below.
+
+### UX Pattern Flow Chart
+
+The following flowchart can be used to determine which UX pattern should be used. This diagram is a useful reference, but may not make sense if you aren't already familiar with the patterns. If this is your first time reading, we recommend you head straight for the [Descriptions of the UX Patterns](#descriptions-of-the-ux-patterns) section.
+
+![New Expensify Data Flow Chart](/web/OfflineUX_Patterns_Flowchart.png)
+
+### How to Answer Questions on the Flow Chart
+
+The numbers in this section correlate to the numbers in each decision box above (the diamond shapes).
+
+1. Does the feature interact with the server?
+
+If you’re changing an existing feature, you can open the network tab of dev tools to see if any network requests are being made when you use the feature. If network requests are being made, the answer to this question is YES.
+If you’re making a new feature, think about whether any data would need to be retrieved or stored from anywhere other than the local device. If data needs to be stored or retrieved, then the answer is YES.
+
+2. What type of request is being made?
+
+This question can be answered by thinking about what’s happening in the server. If there’s new data being saved on the server, you’re making a WRITE request. If you’re retrieving existing data from the server, you’re making a READ request. 
+
+3. Do we need to show up-to-date data?
+
+You should answer yes to this question if it’s important that the user sees current data/content. An example is viewing a chat. Even though the user won’t be able to see new chat messages, it’s still important for us to show the user the old ones.
```

Comment:
I don't think that's a very good example, and specifically in the case of chats, this is not the pattern that would be used. I think maybe a better way to ask this question is... "Is it OK for the user to see stale data?" Example: the payment method list. We don't want the user to see a payment method that we no longer support, not even while the payment methods are being loaded from the server (or while the user is offline). Therefore, we use the blocking UI so that the user doesn't see stale data while we load the payment methods.

Reply:
updated the question and the chart 👍